contracts/api3-server-v1/Api3Market.sol

```diff
@@ -337,6 +333,99 @@ contract Api3Market is HashRegistry, ExtendedSelfMulticall, IApi3Market {
         );
     }
 
+    /// @notice Multi-calls this contract, followed by a call with value to buy
+    /// the subscription
+    /// @dev This function is for the API3 Market frontend to call
+    /// `eth_estimateGas` on a single transaction that readies a data feed and
+    /// buys the respective subscription
+    /// @param multicallData Array of calldata of batched calls
+    /// @param dapiName dAPI name
+    /// @param dataFeedId Data feed ID
+    /// @param sponsorWallet Sponsor wallet address
+    /// @param updateParameters Update parameters
+    /// @param duration Subscription duration
+    /// @param price Subscription price
+    /// @param dapiManagementAndDapiPricingMerkleData ABI-encoded dAPI
+    /// management and dAPI pricing Merkle roots and proofs
+    /// @return returndata Array of returndata of batched calls
+    /// @return subscriptionId Subscription ID
+    function multicallAndBuySubscription(
+        bytes[] calldata multicallData,
+        bytes32 dapiName,
+        bytes32 dataFeedId,
+        address payable sponsorWallet,
+        bytes calldata updateParameters,
+        uint256 duration,
+        uint256 price,
+        bytes calldata dapiManagementAndDapiPricingMerkleData
+    )
+        external
+        payable
+        override
+        returns (bytes[] memory returndata, bytes32 subscriptionId)
+    {
+        returndata = this.multicall(multicallData);
+        subscriptionId = buySubscription(
+            dapiName,
+            dataFeedId,
+            sponsorWallet,
+            updateParameters,
+            duration,
+            price,
+            dapiManagementAndDapiPricingMerkleData
+        );
+    }
+
+    /// @notice Multi-calls this contract in a way that the transaction does
+    /// not revert if any of the batched calls reverts, followed by a call with
+    /// value to buy the subscription
+    /// @dev This function is for the API3 Market frontend to send a single
+    /// transaction that readies a data feed and buys the respective
+    /// subscription. `tryMulticall()` is preferred in the purchase transaction
+    /// because the readying calls may revert due to race conditions.
+    /// @param tryMulticallData Array of calldata of batched calls
+    /// @param dapiName dAPI name
+    /// @param dataFeedId Data feed ID
+    /// @param sponsorWallet Sponsor wallet address
+    /// @param updateParameters Update parameters
+    /// @param duration Subscription duration
+    /// @param price Subscription price
+    /// @param dapiManagementAndDapiPricingMerkleData ABI-encoded dAPI
+    /// management and dAPI pricing Merkle roots and proofs
+    /// @return successes Array of success conditions of batched calls
+    /// @return returndata Array of returndata of batched calls
+    /// @return subscriptionId Subscription ID
+    function tryMulticallAndBuySubscription(
+        bytes[] calldata tryMulticallData,
```

Comment:
It's not nice that this giant argument comes first, but I'm concerned that putting it in the end will be confusing regarding the sequence of the calls